File: app/src/main/java/com/rachelleignacio/listbucket/presentation/presenters/ListItemsFragmentPresenter.java
package com.rachelleignacio.listbucket.presentation.presenters;

import com.rachelleignacio.listbucket.domain.models.ListItem;

/**
 * Created by rachelleignacio on 3/31/17.
 */

public interface ListItemsFragmentPresenter {
    interface View {
        void showListItems(java.util.List<ListItem> items);
        void onItemSwipedToDelete(int position);
    }

    void getListItems();
    int getListCount();
    void addListItem(String listItemName);
    void deleteListItem(int position);
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/presentation/presenters/ListItemsFragmentPresenter.kt
package com.rachelleignacio.listbucket.presentation.presenters

import com.rachelleignacio.listbucket.domain.models.ListItem

/**
 * Created by rachelleignacio on 8/23/17.
 */
interface ListItemsFragmentPresenter {
    interface View {
        fun showListItems(items: MutableList<ListItem>)
        fun onItemSwipedToDelete(adapterPosition: Int)
    }

    fun getListItems()
    val listCount: Int
    fun addListItem(listItemName: String)
    fun deleteListItem(adapterPosition: Int)
}